Cargo.toml

```diff
@@ -24,6 +24,7 @@ tokio = { version = "1.43.0", default-features = false, features = ["signal", "m
 tracing = { version = "0.1.41", default-features = false, features = ["attributes"]}
 tracing-subscriber = { version = "0.3.19", default-features = false, features = ["env-filter", "fmt", "json", "time", "ansi"] }
 url = { version = "2.5.4", default-features = false }
+lru = {version = "0.16.0", default-features = false }
```

Comment:
this dependency can also be useful for resolving another TODO about do not sending duplicate requests to Emily

Reply:
and also if I understand correctly, we will run this binary on machine without any sensitive keys stored on, so new dependencies is not as bad